src/arrayMethodJoin.js

```diff
@@ -4,8 +4,23 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (const item of this) {
+      if (item === null || item === undefined) {
+        string += separator;
+        continue;
+      }
+
+      string += item;
+
+      if (this.lastIndexOf(item) !== this.length - 1) {
```

Comment:
then you don't need lastIndexOf method

```suggestion
      if (i !== this.length - 1) {
```